esp/services/ws_machine/ws_machineService.cpp

```diff
@@ -1495,11 +1495,24 @@ bool CStorageData::read(const char* s)
     if (m_diskSpaceTotal > 0)
         m_diskSpacePercentAvail = round((float)m_diskSpaceAvailable*100/m_diskSpaceTotal);
 
-    //If a given path (ex. /var/lib/HPCCSystems/hpcc-mirror/thor) in the usage request does not exist,
-    //the data.item(2) is the path (ex. /var/lib/HPCCSystems/hpcc-mirror/) the usage script is used
-    //to read the DiskSpace.
-    if (m_readPath && (data.length() > 2))
-        m_path.set(data.item(2));
+    if (data.length() > 2)
+    {
+        pStr = data.item(2);
+        if (isdigit(*pStr))
+        {
+            m_diskSpaceActualSize = atol(pStr);
+            if (m_toMByte)
+                m_diskSpaceActualSize = llround((double)m_diskSpaceActualSize/1024);
+
+            //If a given path (ex. /var/lib/HPCCSystems/hpcc-mirror/thor) in the usage request does not exist,
+            //the data.item(2) is the path (ex. /var/lib/HPCCSystems/hpcc-mirror/) the usage script is used
```

Comment:
This comment needs updating.  To reflect the various meanings of data.item(2), and now data.item(3).